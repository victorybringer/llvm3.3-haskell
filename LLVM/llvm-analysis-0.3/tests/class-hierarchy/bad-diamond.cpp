class Base {
public:
  Base() {}
  virtual void f1() {}
};

class Middle1 : public Base {
public:
  virtual void m1() {}
};

class Middle2 : public Base {
public:
  virtual void m2() const {}
};

class Derived : public Middle1, public Middle2 {
  int m_i;
public:
  Derived(int);
  virtual int f3() const;
};

Derived::Derived(int i): m_i(i) {}
int Derived::f3() const {
  this->m2();
  return m_i;
}
