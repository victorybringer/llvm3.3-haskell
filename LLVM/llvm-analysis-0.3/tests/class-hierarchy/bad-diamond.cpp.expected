fromList [ ("Base", fromList ["Middle1", "Middle2"])
         , ("Middle1", fromList ["Derived"])
         , ("Middle2", fromList ["Derived"])
         ]
